config/src/lib.rs

```diff
@@ -317,7 +322,7 @@ impl ConfigInner {
                     log::trace!("event:{:?}", event);
                     if let Some(path) = extract_path(event) {
                         log::debug!("path {} changed, reload config", path.display());
-                        reload();
+                        reload(ConfigFileSelection::Search);
```

Comment:
I mentioned misbehavior in a comment above; here's a problem scenario:

You set `--config-file` on the command line.  If you update that file, this code triggers and instead of reloading the original file it now loads using the default config file resolution instead of directly loading the file that you specified via the command line option.

For this to be right, we need to remember that the cli override was used.

(It may just happen to squeak by because we also set the environment variable to match what we loaded, but that's more of a lucky coincidence and is the kind of subtle behavior that could get overlooked and broken over time if it isn't explicitly handled)

Reply:
> It may just happen to squeak by because we also set the environment variable to match what we loaded, but that's more of a lucky coincidence and is the kind of subtle behavior that could get overlooked and broken over time if it isn't explicitly handled

Good point you're right, it's better when the behavior is explicit!